compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -617,6 +620,33 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     });
 
                 if let Some((decl, body_id)) = closure_decl_and_body_id {
+                    let mut visitor = UnresolvedTypeFinder::new(self);
+                    for arg in fn_sig.inputs().skip_binder().iter() {
+                        if let ControlFlow::Break((ty, _)) = visitor.visit_ty(*arg) {
+                            let mut inner = self.inner.borrow_mut();
+                            let ty_vars = &inner.type_variables();
+                            let ty_span = match ty.ty_vid() {
+                                Some(ty_vid) => {
+                                    let var_origin = ty_vars.var_origin(ty_vid);
+                                    var_origin.span
+                                }
+                                None => continue,
+                            };
+                            err.span_label(
+                                ty_span,
+                                "consider giving this closure parameter an explicit type without `_` placeholders",
```

Comment:
I prefer a more direct phrasing:

```suggestion
                                "give this closure parameter an explicit type instead of `_`",
```

Reply:
I think it reads a little oddly when the type is something like `Vec<_>` since the way you have it written, it sounds like we think the entire closure parameter currently has the type `_` (and i copied the wording from the pre-existing return value case). But I don't have super strong feelings either way

Reply:
makes sense, then can you at least `s/consider giving/give`?